src/validators/raw_gtfs.rs

```diff
@@ -20,23 +19,16 @@ fn check_duplicates<O: gtfs_structures::Id + gtfs_structures::Type>(
 }
 
 pub fn validate(raw_gtfs: &gtfs_structures::RawGtfs) -> Vec<Issue> {
-    check_duplicates(&raw_gtfs.stops, Severity::Warning)
+    check_duplicates(&raw_gtfs.stops)
         .into_iter()
-        .chain(check_duplicates(&raw_gtfs.routes, Severity::Warning).into_iter())
-        .chain(check_duplicates(&raw_gtfs.trips, Severity::Warning).into_iter())
+        .chain(check_duplicates(&raw_gtfs.routes).into_iter())
+        .chain(check_duplicates(&raw_gtfs.trips).into_iter())
+        .chain(check_duplicates(&raw_gtfs.agencies).into_iter())
+        .chain(check_duplicates(&raw_gtfs.pathways.as_ref().unwrap_or(&Ok(vec![]))).into_iter())
+        .chain(check_duplicates(&raw_gtfs.shapes.as_ref().unwrap_or(&Ok(vec![]))).into_iter())
```

Comment:
Primary key for shapes should be `(shape_id, shape_pt_sequence)` https://gtfs.org/schedule/reference/#shapestxt, sorry I saw it too late!